django/core/management/commands/shell.py

```diff
@@ -111,7 +126,22 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_namespace(self):
+        apps_models = apps.get_models()
+        imported_objects = {}
+        for model in reversed(apps_models):
+            if model.__module__:
+                imported_objects[model.__name__] = model
+        return imported_objects
+
+    def update_globals(self):
+        imported_objects = self.get_namespace()
+        for model_name, model in imported_objects.items():
+            globals()[model_name] = model
```

Comment:
I don’t think we should be updating `globals()` here. That can break the rest of the `handle` method and more, for example, if a user’s custom `get_namespace()` includes something `exec` that isn’t Python’s built-in one.

Instead, create a fresh dictionary for globals, merging in the special namsepace:

```python
exec(options["command"], {**globals(), **self.get_namespace()})
```

...and the same for the `stdin` case.